src/renderer/base/renderer.hpp

```diff
@@ -25,6 +25,8 @@ Author(s):
 #include "../../buffer/out/textBuffer.hpp"
 #include "../../buffer/out/CharRow.hpp"
 
+typedef interval_tree::IntervalTree<til::point, size_t> ThisTree;

```

Comment:
If we just typedef this in `textBuffer.hpp`, then do we need to repeat the typedef in all the other files?

(alternatively: I thought it was best practice to avoid typedefs in headers)